scipy/optimize/_differentialevolution.py

```diff
@@ -1216,10 +1216,11 @@ def solve(self):
 
                 constr_violation = self._constraint_violation_fn(DE_result.x)
                 if np.any(constr_violation > 0.):
-                    warnings.warn("differential evolution didn't find a"
-                                  " solution satisfying the constraints,"
-                                  " attempting to polish from the least"
-                                  " infeasible solution", UserWarning)
+                    warnings.warn("differential evolution didn't find a "
+                                  "solution satisfying the constraints, "
+                                  "attempting to polish from the least "
+                                  "infeasible solution",
+                                  UserWarning, stacklevel=2)
```

Comment:
I think technically this is correct, even though DE is probably mostly used via `differential_evolution`. I've wondered what best practice is in such cases. Let's stick with `2` here; just thought I'd mention it.